synapse/replication/http/_base.py

```diff
@@ -144,19 +150,20 @@ async def _serialize_payload(**kwargs):
         return {}
 
     @abc.abstractmethod
-    async def _handle_request(self, request, **kwargs):
+    async def _handle_request(
+        self, request: Request, **kwargs: Any
+    ) -> Tuple[int, JsonDict]:
         """Handle incoming request.
 
         This is called with the request object and PATH_ARGS.
 
         Returns:
-            tuple[int, dict]: HTTP status code and a JSON serialisable dict
-            to be used as response body of request.
+            HTTP status code and a JSON serialisable dict to be used as response
+            body of request.
         """
-        pass
 
     @classmethod
-    def make_client(cls, hs: "HomeServer"):
+    def make_client(cls, hs: "HomeServer") -> Callable:
```

Comment:
This is pretty weak, but the inputs depend on `_serialize_payload`, so I'm not really sure if there's anything better to do?

Reply:
I think we could do something better, yes. I don't think it is worth holding up the PR for it though. Would adding a TODO help? Or maybe filing an issue?

Reply:
I think mypy will complain (with the right `--option`) about a generic type like `Callable` used without fully specifying its parameters, so I don't think this one needs to be called out in particular.